GitCommands/Git/GitModule.cs

```diff
@@ -2367,9 +2373,12 @@ public void SetPathSetting(string setting, string value)
             LocalConfigFile.SetPathValue(setting, value);
         }
 
-        public IReadOnlyList<GitStash> GetStashes()
+        public IReadOnlyList<GitStash> GetStashes(bool noLocks = false)
         {
-            var args = new GitArgumentBuilder("stash") { "list" };
+            var args = new GitArgumentBuilder("stash", gitOptions:
+                noLocks && GitVersion.Current.SupportNoOptionalLocks
+                    ? (ArgumentString)"--no-optional-locks"
+                    : default) { "list" };
```

Comment:
Make it readable.  Something like below.

```c#
 var args = new GitArgumentBuilder("stash", gitOptions:
                noLocks && GitVersion.Current.SupportNoOptionalLocks
                    ? (ArgumentString)"--no-optional-locks"
                    : default)
                { "list" };
```

Reply:
I used the existing pattern, can change. Should there even be a background option to GitArgumentBuilder?

Reply:
If you are talking about gitOptions, it is there so we can provide https://git-scm.com/docs/git#_options to git.

Reply:
Changing this line

--no-optional-locks is  intended for background commands. (CommitForm frequently had "index locked" before this change in 2.51.03

"noLocks" argument could have been "background" instead, that is what it is used for.

Reply:
GitOptions as I [linked ](https://git-scm.com/docs/git#_options )before provides for ANY git option, not just --no-optional-locks.

Reply:
NoLocks was used to be clear.  It tells the user the fact that locks are not desired.  Background may be the result but nolocks gives more detail.

Reply:
This is not to change now anyway, but this setting should also set the thread priority to a low value, to let interactive commands finish first. The option will then better be renamed.